web/controllers/courseController/index.ts

```diff
@@ -2903,17 +2950,47 @@ async function instructorMail(pupil: Pupil, courseId: number, subcourseId: numbe
         logger.warn("Tried to send instructor mail to invalid subcourse");
         return 404;
     }
-
+    
+    
+    const lectures = subcourse.lectures.sort(
+        (a, b) => a.start.getMilliseconds() - b.start.getMilliseconds()
+    );
+    const lastLecture = lectures[lectures.length - 1];
+    if (lastLecture != null) {
+        const lectureEnd = moment(lastLecture.start)
+            .add(lastLecture.duration, 'minutes');
+        if (moment().isAfter(lectureEnd.add(14, 'days'))) {
+            logger.warn("Tried to send instructor mail after 14 days passed since the course has ended");
+            return 400;
+        }
+    } else {
+        logger.warn("Can't determine the end date of the course");
+        return 400;
+    }
+    
 
     try {
-        // send mail to correspondnet
-        await sendParticipantToInstructorMail(pupil, course.correspondent, course, mailSubject, mailBody);
+        let attachmentGroupId = uuid().toString();
+
+        let attachmentURLs = "";
+
+        if (files.length > 0) {
+            attachmentURLs += "<h3>Anhänge</h3>";
+
+            await Promise.all(files.map(async f => {
+                let attachmentId = await createAttachment(f, pupil, attachmentGroupId);
+                attachmentURLs = attachmentURLs + `<p><a href="https://api2.corona-school.de/api/attachments/${attachmentId}/${f.originalname}">${f.originalname}</a> (${friendlyFileSize(f.size, true)})</p>`
+            }))
+        }
+
+
         await Notification.actionTaken(pupil, "instructor_course_participant_message", {
             participant: pupil,
             course: dropCourseRelations(course),
             subcourse: dropSubcourseRelations(subcourse),
             subject: mailSubject,
-            body: mailBody
+            body: mailBody,
+            attachmentURLs
```

Comment:
I would prefer if `Notification.actionTaken` would take in the attachments not just as "another context" but as a dedicated parameter. That way, we can then also store the attachment group ids in the concrete notification, to trace back which file was shared with whom and when (e.g. when we have to delete some of the files manually for some reason, or if we add "garbage collection" somewhen).